lib/iris/tests/unit/analysis/test_Aggregator.py

```diff
@@ -315,6 +319,39 @@ def test_kwarg_pass_through_combined_kwargs(self):
         expected_kwargs.update(call_kwargs)
         lazy_func.assert_called_once_with(data, axis, **expected_kwargs)
 
+class Test_aggregated_by(tests.IrisTest):
+    def setUp(self):
+        # This sets up a cube with a time coordinate containing two years and
+        # two months.
+        latitude = DimCoord(np.linspace(-90, 90, 4),
+                            standard_name='latitude',
+                            units='degrees')
+        longitude = DimCoord(np.linspace(45, 360, 8),
+                             standard_name='longitude',
+                             units='degrees')
+        time_unit = unit.Unit('days since 1970-01-01 00:00:00',
+                              calendar=unit.CALENDAR_360_DAY)
+        time = DimCoord(np.linspace(0, 750, 26),
+                        standard_name='time',
+                        units=time_unit)
+        self.time_cube = Cube(np.zeros((26, 4, 8), np.float32),
+                               dim_coords_and_dims=[(time, 0),
+                                                    (latitude, 1),
+                                                    (longitude, 2)])
+
+    def test_month_name(self):
+        # This will test the ability of the aggregator to correctly calculate
+        # the order of months in the new coordinate array.
+        # The month names should appear in chronological order from January
+        # through to December.
+        coord_categorisation.add_month_number(self.time_cube, 'time',
+                                              name='month')
+        agg_cube = self.time_cube.aggregated_by('month', MEAN)
+        coord_categorisation.add_month(agg_cube, 'time', name='month_name')
+        self.assertEquals(agg_cube.coord('month_name').points[0], 'Jan')
```

Comment:
No I initially used March because that was the first one that fell through, then I realised that maybe I should spread out the checks a little across months in case I apply a fix which works for March but changes the order later on.  A loop sounds like a good idea though, I will implement that.